ddtrace/internal/datadog/profiling/stack_v2/src/stack_renderer.cpp

```diff
@@ -119,6 +130,13 @@ StackRenderer::render_python_frame(std::string_view name, std::string_view file,
     if (!utf8_check_is_valid(file.data(), file.size())) {
         file = invalid;
     }
+    // DEV: Echion pushes a dummy frame containing task name, and its line
+    // number is set to 0.
+    if (!task_state.pushed_task_name and line == 0 and name != invalid and name != "<invalid>" and file != invalid) {
```

Comment:
Removed `file`. I think I added the checks as I've been getting `<invalid>` for task names in the tests. I don't mind showing more of `<invalid>` ask we're already showing this to the users.

Reply:
Yeah--I don't think there's a wrong way or a right way.

The only quantitative thing here is that omitting the task name leads to a UX where the underlying execution appears as though it did not occur within a task at all.  It's actually been incredibly useful (for, uh... profiler escalations 😢) to know when things are on a task or not--even if the task name is garbage.

Reply:
That's a reasonable concern but it would be better to be implemented in echion. Instead of simply using `<invalid>`, it could use context specific `<invalid>`, i.e. <`invalid-task-name>`